### ISSUE TITLE
resteasy-codestart and resteasy extension added when creating project with resteasy-reactive-jaxb

### ISSUE THREAD

**Issue by jsmrcka** [quarkusio/quarkus#24818]
### Describe the bug

Combination of `resteasy-reactive-jaxb` and `resteasy` then causes build failure:
```
[ERROR] my.groupId.GreetingResourceTest.testHelloEndpoint  Time elapsed: 0.016 s  <<< ERROR!
java.lang.RuntimeException: 
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.deployment.steps.CapabilityAggregationStep#aggregateCapabilities threw an exception: java.lang.IllegalStateException: Please make sure there is only one provider of the following capabilities:
capability io.quarkus.rest is provided by:
  - io.quarkus:quarkus-resteasy-reactive::jar:2.7.5.Final
  - io.quarkus:quarkus-resteasy::jar:2.7.5.Final
	at io.quarkus.deployment.steps.CapabilityAggregationStep.aggregateCapabilities(CapabilityAggregationStep.java:147)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

### Expected behavior

`🚀  resteasy-reactive-codestart` should probably be picked instead of `🚀  resteasy-codestart`.
No `quarkus-resteasy` in POM.

### Actual behavior

```
quarkus create app my-groupId:myapp --extension=resteasy-reactive-jaxb
-----------
selected extensions: 
- io.quarkus:quarkus-resteasy-reactive-jaxb


applying codestarts...
📚  java
🔨  maven
📦  quarkus
📝  config-properties
🔧  dockerfiles
🔧  maven-wrapper
🚀  resteasy-codestart

-----------
```
pom.xml:
```
...
  <dependencies>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy-reactive-jaxb</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-arc</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-junit5</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>
...
```

### How to Reproduce?

```
quarkus create app my-groupId:myapp --extension=resteasy-reactive-jaxb
cd ./myapp/
quarkus build
```

### Output of `uname -a` or `ver`

Linux ... 5.16.12-200.fc35.x86_64

### Output of `java -version`

openjdk version "11.0.14" 2022-01-18 OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9) OpenJDK 64-Bit Server VM Temurin-11.0.14+9 (build 11.0.14+9, mixed mode)

### Quarkus version or git rev

2.7.5.Final, 2.8.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

**Comment by geoand** [comment]
This sounds like one for @ia3andy.

**Comment by ia3andy** [comment]
This is fixed in main (and starting from 2.8 I suppose): https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-reactive/quarkus-resteasy-reactive-jaxb/runtime/src/main/resources/META-INF/quarkus-extension.yaml#L18
